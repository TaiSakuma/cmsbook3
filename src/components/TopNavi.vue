<template>
  <v-tabs optional align-with-title background-color="transparent">
    <v-tab
      v-for="page in pages"
      :key="page.name"
      :to="page.path"
      v-text="page.name"
      class="no-uppercase"
    ></v-tab>
    <v-tab to="/" v-show="false">
      <!-- to hide v-tabs-slider when none is selected -->
    </v-tab>
  </v-tabs>
</template>

<script>
import axios from "axios";

export default {
  name: "TopNavi",
  data: () => ({
    pages: []
  }),
  methods: {
    async updatePages() {
      let path = process.env.VUE_APP_CMSBOOK_URL;
      path = path + "/.cmsbook3/chapters.json";
      try {
        const response = await axios.get(path);
        this.pages = response.data["chapters"];
      } catch (error) {
        this.pages = [];
      }
    }
  },
  created() {
    this.updatePages();
  }
};
</script>

<style scoped>
.no-uppercase {
  text-transform: none;
}
</style>
