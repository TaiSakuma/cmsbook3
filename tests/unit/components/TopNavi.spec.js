import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { mount, shallowMount, createLocalVue } from "@vue/test-utils";

import moxios from "moxios";

import TopNavi from "@/components/TopNavi.vue";

Vue.use(Vuetify);
// Vue.use(VueRouter);

describe("TopNavi.vue", () => {
  const ENV_ORG = process.env;

  let localVue;
  let router;
  let wrapper;

  beforeEach(() => {
    process.env.VUE_APP_CMSBOOK_URL = "http://localhost/cmsbook";
    moxios.install();
    localVue = createLocalVue();
    router = new VueRouter();

    wrapper = shallowMount(TopNavi, {
      localVue,
      router,
      mocks: {
        $route: {
          path: "/chapter-A/section-b/page-3.md",
          params: {
            chapter: "chapter-A",
            section: "section-b",
            page: "page-3.md"
          }
        }
      }
    });
  });

  afterEach(() => {
    moxios.uninstall();
    process.env = ENV_ORG;
  });

  const response = {
    chapters: [
      { name: "Chapter A", path: "/chapter-A" },
      { name: "Chapter B", path: "/chapter-B" },
      { name: "Chapter C", path: "/chapter-C" }
    ]
  };

  it("snapshot", done => {
    moxios.wait(async () => {
      let request = moxios.requests.mostRecent();
      await request.respondWith({
        status: 200,
        response: response
      });
      expect(wrapper.html()).toMatchSnapshot();
      done();
    });
  });

  it("axios request ", done => {
    moxios.wait(() => {
      let request = moxios.requests.mostRecent();
      expect(request.config.url).toBe(
        "http://localhost/cmsbook/.cmsbook3/chapters.json"
      );
      expect(request.config.method).toBe("get");
      expect(request.config.data).toBeUndefined();
      done();
    });
  });

});
