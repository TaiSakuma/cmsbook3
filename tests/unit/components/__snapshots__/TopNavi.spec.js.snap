// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavi.vue snapshot 1`] = `
<v-tabs-stub activeclass="" alignwithtitle="true" backgroundcolor="transparent" mobilebreakpoint="1264" nexticon="$next" optional="true" previcon="$prev" slidersize="2">
  <v-tab-stub to="/chapter-A" ripple="true" class="no-uppercase">Chapter A</v-tab-stub>
  <v-tab-stub to="/chapter-B" ripple="true" class="no-uppercase">Chapter B</v-tab-stub>
  <v-tab-stub to="/chapter-C" ripple="true" class="no-uppercase">Chapter C</v-tab-stub>
  <v-tab-stub to="/" ripple="true" style="display: none;"></v-tab-stub>
</v-tabs-stub>
`;
